### ISSUE TITLE
Properties in a BlockList field are not ignored

### ISSUE THREAD

**Issue by earleskinner-avast** [Jumoo/Jumoo.TranslationManager.Issues#11]
When creating a set, if you choose to ignore a property which resides in a BlockList. It will not ignore it. 

Test:
![image](https://user-images.githubusercontent.com/66742850/124451436-4993b700-dd7d-11eb-9068-1ae0148a583d.png)
The `linkUrl` field is part of a Links Block List. The zip contains a XLIFF translation showing the field still exists.

[20210705_103252_en-US_nl-NL.zip](https://github.com/Jumoo/Jumoo.TranslationManager.Issues/files/6763276/20210705_103252_en-US_nl-NL.zip)

**Comment by KevinJump** [comment]
Hi, 

Yes currently we don't support properties inside properties  :(  - so the property list is only at the top level of a document. (we should update the description!)

the recommendation is to use our Non-Translatable text / text area property editors - which are can be used in place of text/text-area. and because they have different editor aliases, Translation Manager ignores them, see https://www.nuget.org/packages/Jumoo.TranslationManager.NoneTranslatableText/

**Comment by earleskinner-avast** [comment]
Hi. Thanks for the info. Unfortunately, I can't just rip up the site to use new fields and we've got more than just text in the block list fields: Media Pickers, Url Pickers, Custom Properties etc. Is there a way to create a custom handler for these different property types via DI?

**Comment by KevinJump** [comment]
Hi,

yeah just to let you know we have been looking at this today 

there is a way to intercept and change the translation values, by hooking into the events Translation Manager fires, but its not very intuitive or consistent - so we've been tiding that up making sure it fires when expected etc. 

we are also looking at adding a 'path' value to the translations so you can work out the property path of something that is nested (e.g `gridContent/grid/row/doctypegrideditor/myNestedDocType/bodyContent`) this will enable more granular control of properties. 

We might have regex or something - so `bodyContent` would work but so would the more precises `gridContent/.+/bodyContent` would be only remove bodyContent if its in the grid.

hopefully this week we will have something a bit better for removing properties that are nested within other properties (so block, gird, nested content etc). 

The main thing is making sure the UI works and we need to check don't break existing setups. 

---

also just to note: the the non-Translation Text properties are designed to be drop in, you don't need to create new properties, but replace the editor type of the existing ones with the non-translation ones, the data doesn't change and everything does still work.

**Comment by KevinJump** [comment]
Hi, 

fixed this in the latest release https://github.com/Jumoo/Jumoo.TranslationManager.Issues/releases/tag/8.7.0

quick answer is tick the new 'Use RegEx' box underneath the properties of a set, but its actually super powerful and you can use regex to include/exclude things based on what they are nested it and whenever they are if you really want to.

**Comment by earleskinner-avast** [comment]
Hi @KevinJump,

Found any issue with the latest release when turning on the use Regex option:
![image](https://user-images.githubusercontent.com/66742850/125752696-9ec20432-5517-4b71-96c8-f6f2ab98ff8e.png)

From the logs:
```
{"@t":"2021-07-15T08:05:20.3570872Z","@mt":"Error Creating Nodes","@l":"Error","SourceContext":"Jumoo.TranslationManager.Core.Controllers.TranslationNodeApiController","ProcessId":3896,"ProcessName":"w3wp","ThreadId":33,"AppDomainId":2,"AppDomainAppId":"LMW3SVC4ROOT","MachineName":"LONA-000757","Log4NetLevel":"ERROR","HttpRequestNumber":3,"HttpRequestId":"0c11ff23-3a3e-4975-9868-fb9adb660ef9"}
```

**Comment by KevinJump** [comment]
yeah we spotted that too 😞  - just testing the fix for this now.

**Comment by KevinJump** [comment]
Hi fixed in 8.7.1 https://github.com/Jumoo/Jumoo.TranslationManager.Issues/releases/tag/8.7.1

**Comment by earleskinner-avast** [comment]
Thanks Kevin. Just tested this now and everything is working great. Thank you for the speedy responses.
